### PR TITLE
Return the unmatched suffix back to the buffer so that it can be match again

### PR DESCRIPTION
**Pull request by cshung**
This is intended to fix issue #27

### CODE REVIEW

**Comment by cshung** [comment]
@skalle

**Comment by cshung** [comment]
@skalle

**Comment by cshung** [comment]
@skalle

File: expect.go [hunk 1]
```diff
@@ -1096,6 +1099,15 @@ func (e *GExpect) Read(p []byte) (nr int, err error) {
 	return e.out.Read(p)
 }
 
+// Return the unmatched suffix back to the buffer so that it can be matched again
```
Comment (skalle) [inline expect.go]: Comments should be of the format // ReturnUnmatchedSuffix ...

Comment (skalle) [inline expect.go]: Don't thing this method should be exported.

File: expect.go [hunk 2]
```diff
@@ -714,7 +714,10 @@ func (e *GExpect) ExpectSwitchCase(cs []Caser, timeout time.Duration) (string, [
 			}
 
 			// Clear the buffer directly after match.
```
Comment (skalle) [inline expect.go]: This commetn needs to be expanded a bit after the changes happened.

File: expect.go (hunk 1 above)
Comment (cshung) [reply expect.go]: Will be fixed in the next iteration.

Comment (cshung) [reply expect.go]: Will be fixed in the next iteration.

**Comment by cshung** [comment]
In this iteration, I fixed the obvious stuff. Now I know go use casing to determine exports. For non-export functions, it appears to me that the existing convention is to not comment them, so I simply deleted the comment.

> Also tests are needed.

I will try to add one - so `expect` a new iteration that comes with test, pun intended. :)

**Comment by cshung** [comment]
@skalle

**Comment by cshung** [comment]
@skalle

**Comment by skalle** [comment]
Hey Andrew. 

One last thing , could you add an Option to enable/disable this behaviour. 
Might be workflows out there relying on the current way it's done.

**Comment by skalle** [comment]
Hey. Sorry again for the slow replies. 

Thx. a lot for fixing this, making go Expect better for everyone.

**Pull request merged**